FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -680,6 +697,10 @@ ContextImpl::GenerateIR(FEXCore::Core::InternalThreadState* Thread, uint64_t Gue
         if (Thread->OpDispatcher->FinishOp(DecodedInfo->PC + DecodedInfo->InstSize, i + 1 == InstsInBlock)) {
           break;
         }
+
+        while (InstForceTSOIt != ForceTSOInstructions.end() && *InstForceTSOIt < Block.Entry + BlockInstructionsLength) {
+          InstForceTSOIt++;
+        }
```

Comment:
Not entirely happy with how the `InstForceTSOIt` variable is used in just three isolated places that are however spread across 150 lines of code. One minor improvement to this is to at least replace this loop-in-a-loop-in-a-loop:

```suggestion
        InstForceTSOIt = std::find_if(InstForceTSOIt, ForceTSOInstructions.end(), [&](auto Val) { return Val >= Block.Entry + BlockInstructionsLength; });
```

Perhaps it would be better to move this (and the `ForceTSO` declaration) to the top of the inner loop though? That would keep the uses of that variable much more local.

Reply:
This can't really be moved to the top of the loop, it's intended this happens at the end after BlockInstructionsLength is updated and after any breaks that could occur that would make the work redundant

Reply:
I'm fine with it walking the interval list like this

Reply:
Should I revert my change?